chromium_src/content/browser/speech/network_speech_recognition_engine_impl.cc

```diff
@@ -5,46 +5,74 @@
 
 #include <string>
 
-#include "base/feature_list.h"
-#include "base/metrics/field_trial_params.h"
+#include "base/containers/flat_map.h"
+#include "brave/components/brave_service_keys/brave_service_key_utils.h"
 #include "brave/components/constants/brave_services_key.h"
+#include "brave/components/speech_to_text/buildflags.h"
+#include "brave/components/speech_to_text/features.h"
+#include "services/network/public/cpp/resource_request.h"
 #include "url/gurl.h"
 
-namespace content {
-namespace google_apis {
+namespace content::google_apis {
 std::string GetAPIKey() {
   return BUILDFLAG(BRAVE_SERVICES_KEY);
 }
-}  // namespace google_apis
-}  // namespace content
+}  // namespace content::google_apis
 
 namespace {
 
-BASE_FEATURE(kSttFeature, "speech_to_text", base::FEATURE_DISABLED_BY_DEFAULT);
-const base::FeatureParam<std::string> kSttUrl{&kSttFeature, "web_service_url",
-                                              ""};
+std::string GetWebServiceBaseUrl(const char* web_service_base_url_for_tests) {
+  if (base::FeatureList::IsEnabled(stt::kSttFeature)) {
+    return stt::kSttUrl.Get();
+  }
+  // Fallback to for-tests URL.
+  return web_service_base_url_for_tests;
+}
+
+void AddBraveHeaders(network::ResourceRequest* request,
+                     const std::string& request_key) {
+  DCHECK(request && !request->method.empty() && request->url.is_valid());
+
+  const std::string brave_sticky_session_cookie =
+      "Brave-stt-sticky=" + request_key;
+  request->headers.SetHeader(net::HttpRequestHeaders::kCookie,
```

Comment:
@boocmp @hspencer77 have we tested this with an ALB yet? from reading the docs i'm a bit concerned that we need to have the backend set the cookie, otherwise the ALB will not set the required `AWSALBAPP` cookie. [1]

[1] https://repost.aws/knowledge-center/elb-alb-stickiness

Reply:
Yes, you're right. I misunderstood the docs. I'll fix it.